docs/frameworks.rst

```diff
@@ -81,4 +81,36 @@ To generate segment based on incoming requests, you need to instantiate the X-Ra
     XRayMiddleware(app, xray_recorder)
 
 Flask built-in template rendering will be wrapped into subsegments.
-You can configure the recorder, see :ref:`Configure Global Recorder <configurations>` for more details.
\ No newline at end of file
+You can configure the recorder, see :ref:`Configure Global Recorder <configurations>` for more details.
+
+aiohttp Server
+==============
+
+For X-Ray to create a segment based on an incoming request, you need register some middleware with aiohttp. As aiohttp
+is an asyncronous framework, X-Ray will also need to be configured with an ``AsyncContext`` compared to the default threadded
+version.::
+
+    import asyncio
+
+    from aiohttp import web
+
+    from aws_xray_sdk.ext.aiohttp.middleware import middleware
+    from aws_xray_sdk.core.async_context import AsyncContext
+    from aws_xray_sdk.core import xray_recorder
+    # Configure X-Ray to use AsyncContext
+    xray_recorder.configure(service='service_name', context=AsyncContext())
+
+
+    async def handler(request):
+        return web.Response(body='Hello World')
+
+    loop = asyncio.get_event_loop()
+    # Use X-Ray SDK middleware
+    app = web.Application(middlewares=[middleware])
```

Comment:
It is strongly recommended to put the X-Ray middleware at first. This has two benefits. One is that other middleware down the chain could throw 403 or other errors and this can be captured if the incoming request hit X-Ray middleware first. The second one is if other middleware tries to generate any subsegments, the segment must be present. It's better to demonstrate the order and its importance in the example.

Reply:
Would you rather I added a second middleware in the example or just a comment emphasising the importance of that? (as well as adding to the docs)